superset/sql_lab.py

```diff
@@ -245,7 +252,20 @@ def execute_sql_statement(
                 query.id,
                 str(query.to_dict()),
             )
-            data = db_engine_spec.fetch_data(cursor, query.limit)
+            data = db_engine_spec.fetch_data(cursor, increased_query)
+            if query.limit is None or len(data) <= query.limit:
+                query.limiting_factor = LimitingFactor.NOT_LIMITED
+            else:
+                # returning 1 row less then the query.limit
+                data = data[:-1]
+
+    except SoftTimeLimitExceeded as ex:
```

Comment:
Did you add this or is it from a borked merge?

Reply:
This must be a borked merge.